### PR TITLE
Bug 2050005: Fix webpack module federation runtime chunk ID clashes

### PR DESCRIPTION
**Pull request by vojtechszocs**
The webpack-generated remote container, e.g. `plugin-entry.js` contains initialization of a "chunk loading" global array, into which all remote-specific chunks, e.g. `exposed-telemetry-chunk.js` are injected at runtime. (A single chunk can host one or more modules.)

This PR ensures that the "chunk loading" global array identifier uses a unique name suffix (Console plugin name).

Example of change in `plugin-entry.js` for `console-demo-plugin`
```diff
-var chunkLoadingGlobal = self["webpackChunk"] = self["webpackChunk"] || [];
+var chunkLoadingGlobal = self["webpackChunkconsole_demo_plugin"] = self["webpackChunkconsole_demo_plugin"] || [];
```

Example of change in `exposed-telemetry-chunk.js` for `console-demo-plugin`
```diff
-(self["webpackChunk"] = self["webpackChunk"] || []).push([["exposed-telemetry"],{
+(self["webpackChunkconsole_demo_plugin"] = self["webpackChunkconsole_demo_plugin"] || []).push([["exposed-telemetry"],{
```

webpack code ***should*** auto-detect a remote container's unique name from its `package.json` file, however there are two problems with this:

1. webpack code [assumes](https://github.com/webpack/webpack/blob/13a3bc13736e3b7066779623939e9fdaec7bc06b/lib/config/defaults.js#L720) that the `package.json` file is located directly at [context](https://webpack.js.org/configuration/entry-context/#context) directory:
```ts
const pkgPath = path.resolve(context, "package.json");
```
in case of `console-demo-plugin` that would be `/path/to/console/dynamic-demo-plugin/src/package.json` which is obviously wrong. IMO this is a webpack bug; the code should start at context directory but look upwards until it finds a `package.json` file.

2. using `name` from `package.json` file is generally not what we want - instead, we should use `consolePlugin.name` which represents the name of the dynamic plugin (this value should be equal to `metadata.name` of the corresponding `ConsolePlugin` resource)

This PR also updates Console dynamic plugin SDK & dynamic demo plugin to use the latest webpack 5 version.

cc @christianvogt @spadgett

### CODE REVIEW

File: frontend/packages/console-dynamic-plugin-sdk/src/webpack/ConsoleRemotePlugin.ts
Comment (spadgett) [inline frontend/packages/console-dynamic-plugin-sdk/src/webpack/ConsoleRemotePlugin.ts:67]: Was it not necessary to use the initialize hook?

**Review by spadgett — COMMENTED** [review]
Nice work 👍 

Will this cause any errors using a plugin built with an older console remote webpack plugin on a 4.11 cluster?

**Comment by vojtechszocs** [comment]
> Will this cause any errors using a plugin built with an older console remote webpack plugin on a 4.11 cluster?

@spadgett Dynamic plugins built with non-patched webpack `ConsoleRemotePlugin` may still run into chunk loading issues with other plugins (less likely) and/or Console app itself (more likely).

The reason it's more likely for Console app is because the amount of webpack-generated Console app chunks is very high, which increases the risk of chunk ID collisions.

To mitigate the issue, all dynamic plugin should issue a fresh (Console 4.11 / latest compatible) build once this PR is merged and the relevant SDK package is published to npmjs.

**Comment by spadgett** [comment]
@vojtechszocs Right, so my question is specifically whether plugins written for 4.10 using `"@openshift-console/dynamic-plugin-sdk-webpack": "0.0.3"` will still load in 4.11 with this change? (Even if there is a risk of IDs clashing.)

Also will plugins using the newer `@openshift-console/dynamic-plugin-sdk-webpack` work with 4.10?

**Comment by vojtechszocs** [comment]
> @vojtechszocs Right, so my question is specifically whether plugins written for 4.10 using `"@openshift-console/dynamic-plugin-sdk-webpack": "0.0.3"` will still load in 4.11 with this change? (Even if there is a risk of IDs clashing.)

Yes, they should still load.

- `@openshift-console/dynamic-plugin-sdk-webpack` version `0.0.3` depends on `"webpack": "^5.53.0"`
- `@openshift-console/dynamic-plugin-sdk-webpack` version `0.0.4` (latest) depends on `"webpack": "^5.62.1"`

webpack 5.x API changes should be backwards compatible.

`0.0.3` vs. `0.0.4` actual code change diff
- Bug 2017130: omit extension when code ref resolution fails
- Bug 2017050: CONSOLE-2963: Update dynamic plugin SDK to latest webpack
- Bug 2020428: Adapt webpack 5 related code in dynamic plugin SDK

I think that even without ^^ changes, dynamic plugins should still load in 4.11 Console.

> Also will plugins using the newer `@openshift-console/dynamic-plugin-sdk-webpack` work with 4.10?

Yes, they should still work. The webpack runtime module/chunk machinery, like `__webpack_require__` global, should be the same.

**Review by spadgett — APPROVED** [review]
/lgtm
/hold
@christianvogt did you want to validate the fix before we merge?

**Comment by christianvogt** [comment]
I haven't ran this PR change myself. But if the plugins are loading successfully then the change is good with me.

I believe we have yet to _actually_ test whether this is a true fix what we saw in devsandbox. This will still need to be tested in the environment if that's the case. But that would be separate of whether or not this change functions.

**Comment by vojtechszocs** [comment]
> I believe we have yet to _actually_ test whether this is a true fix what we saw in devsandbox.

Yes, that's what we should do. Switch to the commit of telemetry plugin that exposed the problem, modify its `package.json` to consume the appropriate version of `@openshift-console/dynamic-plugin-sdk-webpack` and rebuild, deploy on cluster and if the problem cannot be reproduced (should be retried multiple times) then the fix should be marked as verified.

**Comment by christianvogt** [comment]
@vojtechszocs while testing the specific bug we encountered on devsandbox is something we will do. IMO this change doesn't require the test in order to be approved. As long as plugins continue to function as expected we can move forward with this PR.

**Pull request merged**
